File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int cnt = 0, result = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                cnt++;
            } else if (s.charAt(i) == ')') {
                cnt--;
            }
            result = Math.max(result, cnt);
        }

        return result;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
class Solution:
    def maxDepth(self, s: str) -> int:
        stack = list()

        for letter in s:
            if letter == '(':
                stack.append(letter)
            elif letter == ')':
                depth = 0
                while stack[-1] != '(':
                    depth = max(depth, stack.pop())
                stack.pop()
                stack.append(depth + 1)
            else:
                stack.append(0)

        return max(stack)